plugins/modules/nautobot_server.py

```diff
@@ -0,0 +1,356 @@
+
+
+def main():
+
+    specific_boolean_params = (
+        "clear",
+        "failfast",
+        "skip",
+        "merge",
+        "link",
+        # collectstatic
+        "no-post-process",
+        "no-default-ignore",
+        # makemigrations
+        "empty",
+        # post_upgrade
+        "no-clearsessions",
+        "no-collectstatic",
+        "no-invalidate-all",
+        "no-migrate",
+        "no-remove-staled-content-type",
+        "no-trace-paths",
+    )
+
+    commands_with_noinput = (
+        "createsuperuser",
+        "migrate",
+        "makemigrations",
+        "collectstatic",
+    )
+
+    end_of_command_params = {
+        "migrate": ["app_label", "migration_name"],
+        "makemigrate": ["app_label"],
+    }
+
+    required_if = [
+        ["command", "createsuperuser", ["args"]],
+    ]
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            command=dict(
+                required=True,
+                type="str",
+                choices=[
+                    "createsuperuser",
+                    "migrate",
+                    "makemigrations",
+                    "collectstatic",
+                    "post_upgrade",
+                ],
+            ),
+            args=dict(required=False, type="dict", default={}),
+            project_path=dict(
+                default="/opt/nautobot",
+                required=False,
+                type="path",
+                aliases=["app_path", "chdir"],
+            ),
+            pythonpath=dict(
+                default=None, required=False, type="path", aliases=["python_path"]
+            ),
+            virtualenv=dict(
+                default=None, required=False, type="path", aliases=["virtual_env"]
+            ),
+            db_password=dict(required=True, type="str", no_log=True),
+        ),
+        required_if=required_if,
+    )
+
+    command = module.params["command"]
+    args = module.params["args"]
+    project_path = module.params["project_path"]
+    virtualenv = module.params["virtualenv"]
+
+    _ensure_virtualenv(module)
+
+    cmd = "nautobot-server %s" % (command,)
+
+    environ_vars = {}
+    if project_path:
+        environ_vars["NAUTOBOT_ROOT"] = project_path
+
+    db_password = module.params["db_password"]
+    if db_password:
+        environ_vars["NAUTOBOT_DB_PASSWORD"] = db_password
+
+    if command in commands_with_noinput:
+        cmd = "%s --noinput" % cmd
+
+    for param in args:
+        if args[param] and param in specific_boolean_params:
+            cmd = "%s --%s" % (cmd, param)
+        elif args[param] and param not in end_of_command_params:
+            cmd = "%s --%s=%s" % (cmd, param, args[param])
+
+    # Positional parameters are added at the end of the command, in specific order
+    if command in end_of_command_params:
+        for param in end_of_command_params[command]:
+            if param in args:
+                cmd = "%s %s" % (cmd, args[param])
+
```

Comment:
```suggestion
for arg, value in args.items():
    cmd += " --%s=%s" % (arg, value)
for flag in flags:
    cmd += " --%s" % (flag,)
for arg in optional_args:
    cmd += " %s" %(arg,)
```

adjusting for addition of flags and optional_args params